src/blocks/PagePartialBlock/PagePartialBlock.astro

```diff
@@ -11,17 +12,20 @@ const { block } = Astro.props;
 
 // TODO: only allow one page partial per block
 const pagePartials = await Promise.all(
-  block.items.map(async (item) => {
-    const pagePartial = await getEntry('PagePartials', item.id);
-    return pagePartial?.data;
-  })
+  block.items.map(async (item) => await getEntry('PagePartials', item.id))
 ).then(items => items.filter(pagePartial => pagePartial !== undefined && pagePartial !== null));
+
 ---
+{/* Subscribe to each PagePartial, revist when only one PagePartial is supported */}
+{ pagePartials.map(({ subscription }) => (
+  <PreviewModeSubscription {...subscription} />
+)) }
+
```

Comment:
There's another PR regarding simplifying the PagePartial. That will cut down on the many array functions in this file